xmlschema/validators/groups.py

```diff
@@ -711,7 +711,7 @@ def iter_encode(self, element_data, validation='lax', **kwargs):
                     errors.append((index - cdata_index, particle, occurs, expected))
                 break
             else:
-                if losslessly:
```

Comment:
Hi,
I think this line is correct. A converter is *losslessly* when it converts all the data and maintains the XML ordering and structure, like JsonML. In other case a strict validated encoding may not be possible (it depends by the structures of the groups), so additional checks have to be done before emitting an error. In those cases validation modes 'strict' and 'lax' are different only because the error is raised or collected.

The point is that the model is broken and I think there is not a simply manner to validate the remaining items on occurrences using the ModelVisitor as is. Probably for encoding not losslessly data formats we need a ModelVisitor subclass (at least with a different *advance* method) that handles partial ordering for sequence models (like an all model if the tag is not the same).

Before doing this, for solving simple cases like your sample case, an heuristic can be to find a *different* element that has the same tag. If not found one an error is emitted. 
Thanks

Reply:
Thanks for the reply.

I don't need this for any particular schema that I've come across (so no need for a work around) but I think that it is important that when `validation="strict"` we only return xml that conforms to the xsd and raise errors in all other cases. If users still want to generate the incorrect xml, they can use the solution you outlined in https://github.com/sissaschool/xmlschema/issues/69 and set `validation="lax"`. This patch alerts the user that there may be something wrong and requires them to change their approach (eg. pick a different converter) or explicitly acknowledge the xml may not be ok.

Reply:
Ok, I've understand your point for this. After we can try to integrate this with a better solution for unordered or collapsed elements.
Thanks